project4/3700crawler

```diff
@@ -69,17 +84,30 @@ class WebCrawler:
         path = '/' if not url.path else url.path
         if url.query:
             path += '?' + url.query
-       
-        # Build the request
-        msg = 'GET ' + path + ' HTTP/1.1\nHost: ' + host
-        if self.cookie:
-            msg += 'Cookie: ' + self.cookie
-        msg += CRLF
 
         # Send the request
         sock = self.connect(host)
-        sock.send(str.encode(msg))
-              
+        sock.send(str.encode(make_http('GET', path, {'cookie': self.cookie, 'host':host})))
+
+        # Get the response
+        resp = self.get_response(sock)
+
+        return resp
+
+    def POST(self, url, body):
+        """Implements the HTTP POST method"""
+        # Parse the URL and retrieve the necessary fields
+        url = urlparse(url)
```

Comment:
Since parsing the url (lines 100-104) is the same code in both `GET` and `POST`, let's create a separate `parse_url(url)` function that returns the parsed url.